cpp/src/rolling/rolling_detail.cuh

```diff
@@ -895,6 +899,235 @@ struct rolling_window_launcher {
       stream,
       mr);
   }
+
+  template <typename PrecedingIter, typename FollowingIter>
+  std::unique_ptr<column> get_collect_list_offsets(column_view const& input,
+                                                   PrecedingIter preceding_begin,
+                                                   FollowingIter following_begin,
+                                                   size_type min_periods,
+                                                   rmm::cuda_stream_view stream,
+                                                   rmm::mr::device_memory_resource* mr)
+  {
+    using namespace cudf;
+
+    // Materialize offsets column.
+    auto size_data_type = data_type{type_to_id<size_type>()};
+    auto sizes =
+      make_fixed_width_column(size_data_type, input.size(), mask_state::UNALLOCATED, stream, mr);
+    auto mutable_sizes = sizes->mutable_view();
+    thrust::transform(thrust::device,
+                      preceding_begin,
+                      preceding_begin + input.size(),
+                      following_begin,
+                      mutable_sizes.begin<size_type>(),
+                      [min_periods] __device__(auto preceding, auto following) {
+                        return (preceding + following) < min_periods ? 0 : (preceding + following);
+                      });
+    return strings::detail::make_offsets_child_column(
+      sizes->view().begin<size_type>(), sizes->view().end<size_type>(), stream, mr);
+  }
+
+  template <typename PrecedingIter, typename FollowingIter>
+  std::pair<rmm::device_buffer, size_type> get_collect_list_null_mask(
+    column_view const& input,
+    PrecedingIter preceding_iter,
+    FollowingIter following_iter,
+    size_type min_periods,
+    rmm::cuda_stream_view stream,
+    rmm::mr::device_memory_resource* mr)
+  {
+    rmm::device_buffer null_mask;
+    size_type null_count;
+    std::tie(null_mask, null_count) = valid_if(
+      thrust::make_counting_iterator<size_type>(0),
+      thrust::make_counting_iterator<size_type>(input.size()),
```

Comment:
input column view is just being used for its size, should we just pass size?

Reply:
You're right. That makes sense.